includes/components/class-attribute.php

```diff
@@ -1860,4 +1863,25 @@ public function disable_jetpack_search( $enabled, $query ) {
 
 		return $enabled;
 	}
+
+	/**
+	 * Add archive page redirect.
+	 */
+	public function add_archive_redirect() {
+		foreach ( $this->get_models() as $model ) {
+			if ( is_post_type_archive( hp\prefix( $model ) ) ) {
+
+				// Get page ID.
+				$page_id = absint( get_option( hp\prefix( 'page_' . $model . 's' ) ) );
+
+				if ( ! $page_id ) {
+					continue;
+				}
+
+				// Redirect to custom archive page.
+				wp_redirect( get_permalink( $page_id ) );
```

Comment:
always use wp_safe_redirect if there's no chance it's an external URL, for security